### PR TITLE
Added CSS modules

### PR DESCRIPTION
**Pull request by somya51p**
Added CSS modules. @jennydaman @mairin  Please review the PR and suggest changes if any.
This PR is associated to issue number #151.

### CODE REVIEW

File: src/components/CreatePlugin/CreatePlugin.js [hunk 1]
```diff
@@ -321,17 +321,17 @@ class CreatePlugin extends Component {
...
-      <div className="createplugin">
-        <section className="createplugin-form-section">
-          <div className="createplugin-header">
-            <div className="row">
-              <div className="createplugin-header-container">
-                <div className="createplugin-display-1">
+      <div className={styles['createplugin']}>
+        <section className={styles['createplugin-form-section']}>
+          <div className={styles['createplugin-header']}>
+            <div className={`row`}>
```
Comment (jennydaman) [inline src/components/CreatePlugin/CreatePlugin.js:327]: Would it be possible to incorporate [`@patternfly/react-styles`](https://www.npmjs.com/package/@patternfly/react-styles)?

File: src/components/Dashboard/components/DashTeamView/DashTeamView.js [hunk 2]
```diff
@@ -170,7 +170,7 @@ class DashTeamView extends Component {
                   </Table.Button>
                 </Table.Actions>,
                 <Table.Actions key="1">
-                  <Table.DropdownKebab id="myKebab" pullRight>
+                  <Table.DropdownKebab id={styles['myKebab']} pullRight>
```
Comment (jennydaman) [inline src/components/Dashboard/components/DashTeamView/DashTeamView.js:173]: It might be useful to change this ID to a class.

**Review by jennydaman — COMMENTED** [review]
A lot of good work here. Let me know what you think about [`@patternfly/react-styles`](https://www.npmjs.com/package/@patternfly/react-styles)

File: src/components/Dashboard/components/DashTeamView/DashTeamView.js (hunk 2 above)
Comment (somya51p) [reply src/components/Dashboard/components/DashTeamView/DashTeamView.js:173]: @jennydaman Yeah sure I will replace the ID with class. 

But for [@patternfly/react-styles](https://www.npmjs.com/package/@patternfly/react-styles) not much documentation is available. It's difficult to go for that. But if you wish I can give it a try.

**Comment by somya51p** [comment]
@jennydaman Please have a look. I have converted `id` to `classname` where id were not involved in further code. Please review and suggest changes if any.

**Comment by jennydaman** [comment]
@somya51p I just merged a pretty big PR, #161 — it's created some conflicts. You did not give me write access to your fork, so could you try to resolve the merge conflicts?

After that, let's merge this PR. We can save [@patternfly/react-styles](https://www.npmjs.com/package/@patternfly/react-styles) to be work for another day. I was unable to find much documentation on it either, but it's probably something that could be figured out using an IDE's autocompletion.

**Comment by somya51p** [comment]
@jennydaman I have resolved the conflicts. You may have a look!

**Comment by jennydaman** [comment]
Did something happen between your comment and the force push? The conflicts are still there.

**Comment by somya51p** [comment]
@jennydaman Resolved them all. I don't know what happened, maybe I did something by mistake. I apologize from my end :)

File: src/components/CreatePlugin/CreatePlugin.js (hunk 1 above)
Comment (jennydaman) [reply src/components/CreatePlugin/CreatePlugin.js:328]: Continuing in https://github.com/FNNDSC/ChRIS_store_ui/issues/167

File: src/components/SignIn/SignIn.js [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
+  Card, CardBody, Alert, Button
 } from 'patternfly-react';
-import Button from '../Button';
 import StoreClient from '@fnndsc/chrisstoreapi';
-import './SignIn.css';
+import styles from './SignIn.module.css';
 import chrisLogo from '../../assets/img/chris_logo-white.png';
 import ChrisStore from '../../store/ChrisStore';
 import FormInput from '../FormInput';
 import { Form, Alert, AlertActionCloseButton } from '@patternfly/react-core';
```
Comment (jennydaman) [inline src/components/SignIn/SignIn.js:12]: Does not compile. The merge conflict resolution should remove `Alert` and `Button` imported from `patternfly-react`

#142 #165

**Review by jennydaman — CHANGES_REQUESTED** [review]

File: src/components/SignIn/SignIn.js (hunk 3 above)
Comment (somya51p) [reply src/components/SignIn/SignIn.js:12]: @jennydaman made changes. Please review.

**Comment by jennydaman** [comment]
@somya51p Big sorry for more conflicts! It really seems like we're going back and forth, though these new changes were vital. I've forward-ported some features from master, such as #139 and #119

**Comment by somya51p** [comment]
@jennydaman I have resolved the conflicts now. Please have a look.

File: src/components/Navbar/Navbar.js [hunk 4]
```diff
@@ -69,15 +69,15 @@ const LoginButton = ({ store }) => (
   <NavLink to="/signin">
     <Button
       variant="primary"
-      className="login-button"
+      className={styles['login-button']}
```
Comment (jennydaman) [inline src/components/Navbar/Navbar.js:72]: image

Color is wrong, could have something to do with the patternfly CSS variables.

File: src/components/Welcome/components/WelcomeCategories/WelcomeCategories.js [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
+import styles from "./welcomeCategories.module.css";
 import WelcomeCategory from "../WelcomeCategory/WelcomeCategory";
 
 const WelcomeCategories = () => {
   const [categories, setCategories] = useState(sampleCategories);
 
   return (
-    <div className="welcome-categories">
-      <div className="welcome-categories-header">
+    <div className={styles['welcome-categories']}>
```
Comment (jennydaman) [inline src/components/Welcome/components/WelcomeCategories/WelcomeCategories.js:10]: image

Styling incorrect.

Hmm, reminds me of https://github.com/FNNDSC/ChRIS_store_ui/pull/157#pullrequestreview-640998000

Not sure if it's a bug in the CSS module or incorrect merge conflict resolution.

**Review by jennydaman — CHANGES_REQUESTED** [review]

File: src/components/Welcome/components/WelcomeCategories/WelcomeCategories.js (hunk 5 above)
Comment (somya51p) [reply src/components/Welcome/components/WelcomeCategories/WelcomeCategories.js:10]: @jennydaman I have tried many things but don't know why this is happening in CSS modules. If I am using normal CSS then the cards are responsive so the merge conflict resolution issue might not be there.

**Comment by somya51p** [comment]
@jennydaman You may have a look. We are using `button` components from two different files so CSS modules are not responding if we are using that. I have removed CSS modules for the `Navbar` and `Welcomecategory.` Now it is looking as before. And the problem CSS modules are causing I will soon try to rectify. I request you to merge at least these changes, I will send another PR for those changes else merge conflicts occur.

**Comment by somya51p** [comment]
@jennydaman Please review the changes once. And if possible merge the PR, I will send another PR for the `WelcomeCategory.jsx` and `button` to change to CSS modules.

File: src/components/Dashboard/components/DashTeamView/DashTeamView.js [hunk 6]
```diff
@@ -168,7 +168,7 @@ class DashTeamView extends Component {
                   </Table.Button>
                 </Table.Actions>,
                 <Table.Actions key="1">
-                  <Table.DropdownKebab id="myKebab" pullRight>
+                  <Table.DropdownKebab className={styles['myKebab']} pullRight>
```
Comment (jennydaman) [inline src/components/Dashboard/components/DashTeamView/DashTeamView.js:171]: this class doesn't exist, let's delete it.

File: src/components/Navbar/components/Search/Search.js [hunk 7]
```diff
@@ -88,27 +53,33 @@ const Search = (props) => {
...
         {showAutoComplete && autoCompleteData && autoCompleteData.length > 0 && (
-          <ul  className="ws-global-search-autocomplete">
+          <ul  className={styles['ws-global-search-autocomplete']}>
             {autoCompleteData.map((item, id) => (
               <li
                 key={id}
-                className={id === cursorState ? 'active' : ''}
                 data-id={item.id}
-                onMouseDown={(e) => {onSearch(item.name, 'ENTER')}}
+                onMouseDown={(e) => {
```
Comment (jennydaman) [inline src/components/Navbar/components/Search/Search.js:78]: Your branch contains old commits, could you make sure you've merged `pf4`?

https://github.com/FNNDSC/ChRIS_store_ui/pull/171/files#diff-9a24940a3a0dc9528f50eadd792809d7cf70188d676001f94d7d87d826280a90R78

**Review by jennydaman — CHANGES_REQUESTED** [review]

File: src/components/Navbar/components/Search/Search.js (hunk 7 above)
Comment (somya51p) [reply src/components/Navbar/components/Search/Search.js]: @jennydaman I have made the changes suggested by you and I have rebased my branch with the `pf4` again. Just have a look, now I think everything is up-to-date.

**Review by jennydaman — APPROVED** [review]
A monumental advancement, great job @somya51p

File: src/components/SignIn/SignIn.js [hunk 8]
```diff
@@ -143,16 +143,18 @@ export class SignIn extends Component {
...
                   autoComplete="current-password"
-                  className="signin-password-form-group"
+                  className={styles['signin-password-form-group']}
                 />
                 <Button
-                  className="signin-login-btn"
-                  variant="primary"
-                  loading={loading}
-                  onClick={this.handleSubmit}>
+                  className={styles['signin-login-btn']}
```
Comment (DivyanshiSingh) [inline src/components/SignIn/SignIn.js:149]: @jennydaman @somya51p wrong code merged while resolving conflicts

**Pull request merged**